fdbclient/include/fdbclient/BlobGranuleCommon.h

```diff
@@ -55,6 +56,80 @@ struct GranuleDeltas : VectorRef<MutationsAndVersionRef> {
 	}
 };
 
+// Deltas in key order
+
+// For key-ordered delta files, the format for both sets and range clears is that you store boundaries ordered by key.
+// Each boundary has a corresponding key, zero or more versioned updates (ValueAndVersionRef), and optionally a clear
+// from keyAfter(key) to the next boundary, at a version.
+// In this form, one or more streams of delta boundaries can be merged with a snapshot to efficiently reconstruct the
+// rows at a desired version.
+// The concept of this versioned mutation boundaries is repurposed directly from a prior version of redwood, back when
+// it supported versioned data.
+struct ValueAndVersionRef {
+	Version version;
+	MutationRef::Type op; // only set/clear
+	ValueRef value; // only present for set
+
+	// clear constructor
+	ValueAndVersionRef() {}
```

Comment:
nit: Mapping parameterized or default constructor to specific usecases seems bit error-prone; how about we convert them to `static initXX` methods which can be named more explicitly?